### PR TITLE
Feature anovas

### PR DESCRIPTION
**Pull request by MikeLippincott**
Here I conduct anova analysis to determine which morphology features are most significant

### CODE REVIEW

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+# ## Anova + Post Hoc testing (tukeyHSD)
+
+# In[7]:
+
+
+# anova for each feature in the dataframe with posthoc tukey test to determine which groups are different from each other
+lst = []
+for i in cp_features:
+    formula = f"{i} ~ C(oneb_Metadata_Treatment_Dose_Inhibitor_Dose) + C(Metadata_number_of_singlecells)"
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:104]: What is this formula? I haven't seen this before and curious what this is doing.

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    model = ols(formula, combined_df).fit()
+    aov_table = sm.stats.anova_lm(model, typ=2)
+    # posthoc = MultiComparison(df_DMSO_thapsi[i], df_DMSO_thapsi['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'])
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:107]: Recommend removing commented out code.

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+
+
+# ## Venn Diagrams
+
+# #### Venn diagram prep
+
+# In[10]:
+
+
+# get all group1 rows that are DMSO and group2 rows that are LPS 100
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:163]:
```suggestion
# get all group1 rows that are DMSO and group2 rows that are LPS treatment
```
Based on the sh file for this script, it seems like group2 rows are any dose for LPS treatment.

Comment is the same for the below.

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+tukey_df_sig_trt_1v2 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment1}")
+    & (tukey_df_sig["group2"] == f"{treatment2}")
+]
+# get all group1 rows that are DMSO and group2 rows that are Thapsigargin 10
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:168]:
```suggestion
# get all group1 rows that are DMSO and group2 rows that are Thapsigargin treatment
```

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+tukey_df_sig_trt_1v3 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment1}")
+    & (tukey_df_sig["group2"] == f"{treatment3}")
+]
+# get all group1 rows that are LPS 100 and group2 rows that are Thapsigargin 10
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:173]:
```suggestion
# get all group1 rows that are LPS treatment and group2 rows that are Thapsigargin treatment
```

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+tukey_df_sig_trt_2v3 = tukey_df_sig[
+    (tukey_df_sig["group1"] == f"{treatment2}")
+    & (tukey_df_sig["group2"] == f"{treatment3}")
+]
+
+
+# In[11]:
+
+
+from matplotlib import rcParams
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:183]: Recommended that all imports are at the beginning of the file.

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+plt.subplots_adjust(left=0.15)
+plt.savefig(
+    f"{save_path}/{treatment1}_vs_{treatment2}_number_sig_overlaping_features_per_organelle.png",
+    dpi=300,
+    bbox_inches="tight",
+)
+plt.show()
+
+
+# In[ ]:
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:213]: Recommend removing any unnecessary code blocks. Looks like this wasn't ran so probably is an extra block in the notebook.

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+
+# In[13]:
+
+
+# create a venn diagram of the features that are significant in all conditions
+from matplotlib_venn import venn2, venn3_unweighted
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/scripts/5.anova.py:220]: Same as above comment, recommended that all imports are at the beginning of the file.

File: 1.Exploratory_Data_Analysis/anova_run.sh [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+#SBATCH --partition=amem
+#SBATCH --qos=mem
+#SBATCH --time=24:00:00
+#SBATCH --output=sample-%
+
+module load anaconda
+
+conda activate Interstellar
+
+treatment1s=( DMSO_0.100_DMSO_0.025 )
```
Comment (jenna-tomkinson) [inline 1.Exploratory_Data_Analysis/anova_run.sh:14]: Interesting, just to confirm, you don't need to include commas between treatments right? The space between them is what separates them?

**Review by jenna-tomkinson — APPROVED** [review]
Great PR! I made comments for you to address prior to merging. Some are questions so I can better understand this process, so I would appreciate the responses so I can learn! 😄 
Nice job!

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py (hunk 7 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:213]: fixed

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py (hunk 2 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:107]: removed

File: 1.Exploratory_Data_Analysis/anova_run.sh (hunk 9 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/anova_run.sh:14]: @jenna-tomkinson yes, the command line is a different animal. Weird right?

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py (hunk 1 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:104]: @jenna-tomkinson The formula is the linear model I am using to fit the data to. This allows for variates and co-variates to be used to calculate <ins>**A**</ins> <ins>**N**</ins>umber <ins>**O**</ins>f  <ins>**VA**</ins>riances (ANOVA)

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+]
+trt3 = df[
+    df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].str.contains(f"{treatment3}")
+]
+
+
+# In[5]:
+
+
+# df_LPS_100 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('LPS_100.000_DMSO_0.025')]
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:70]: removed

File: 1.Exploratory_Data_Analysis/scripts/0.nomic_cleanup.py [hunk 11]
```diff
@@ -80,6 +80,19 @@ def add_trailing_zeros(x):
...
 
 
+nomic_df["Metadata_inhibitor"].replace("Media ctr", "Media_ctr", inplace=True)
+nomic_df["Metadata_inducer1"].replace("media ctr", "Media_ctr", inplace=True)
+
+
+# In[8]:
+
+
+nomic_df.replace("nan", np.nan, inplace=True)
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/0.nomic_cleanup.py:90]: I think the nomic dataset had `'nan'` strings and not np.nans

File: 1.Exploratory_Data_Analysis/scripts/0.nomic_cleanup.py
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/0.nomic_cleanup.py:1]: might have reviewed prior to a push but its is in order here

File: 1.Exploratory_Data_Analysis/ 
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/ :1]: removed weird file

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 12]
```diff
@@ -0,0 +1,385 @@
...
+)
+print(tukey_df_sig_trt_1v2_unique["organelle"].unique())
+
+
+# #### Plot number of significant features by organelle
+
+# In[17]:
+
+
+# drop the other organelle
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:345]: No reason, I should Look into this though....

File: 1.Exploratory_Data_Analysis/scripts/5.anova.py [hunk 13]
```diff
@@ -0,0 +1,385 @@
...
+tukey_df_sig_trt_1v2_unique["organelle"] = tukey_df_sig_trt_1v2_unique[
+    "features"
+].str.split("_", expand=True)[3]
+tukey_df_sig_trt_1v2_unique["organelle"].unique()
+
+
+# In[16]:
+
+
+tukey_df_sig_trt_1v2_unique["organelle"] = tukey_df_sig_trt_1v2_unique[
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/scripts/5.anova.py:297]: implemented and it looks good!

**Pull request merged**